src/olympia/abuse/models.py

```diff
@@ -1018,3 +866,158 @@ def target(self):
+        return {
+            DECISION_ACTIONS.AMO_BAN_USER: CinderActionBanUser,
+            DECISION_ACTIONS.AMO_DISABLE_ADDON: CinderActionDisableAddon,
+            DECISION_ACTIONS.AMO_REJECT_VERSION_ADDON: CinderActionRejectVersion,
+            DECISION_ACTIONS.AMO_REJECT_VERSION_WARNING_ADDON: (
+                CinderActionRejectVersionDelayed
+            ),
+            DECISION_ACTIONS.AMO_ESCALATE_ADDON: CinderActionEscalateAddon,
+            DECISION_ACTIONS.AMO_DELETE_COLLECTION: CinderActionDeleteCollection,
+            DECISION_ACTIONS.AMO_DELETE_RATING: CinderActionDeleteRating,
+            DECISION_ACTIONS.AMO_APPROVE: CinderActionApproveInitialDecision,
+        }.get(decision_action, CinderActionNotImplemented)
+
+    def get_action_helper(self, existing_decision=None, *, override=False):
+        # Base case
+        CinderActionClass = self.get_action_helper_class(self.action)
+        # But we use more specific actions for certain cases:
+        # Where there was an appeal/override from a remove action to approve
+        if self.action == DECISION_ACTIONS.AMO_APPROVE and (
+            existing_decision in DECISION_ACTIONS.REMOVING
+        ):
+            CinderActionClass = (
+                CinderActionOverrideApprove
+                if override
+                else CinderActionTargetAppealApprove
+            )
+
+        # Where there was an appeal/override which didn't change from a remove action
+        elif self.action == existing_decision and (
+            self.action in DECISION_ACTIONS.REMOVING
+        ):
+            CinderActionClass = (
+                CinderActionNotImplemented
+                if override
+                else CinderActionTargetAppealRemovalAffirmation
+            )
+
+        return CinderActionClass(decision=self)
+
+    def can_be_appealed(self, *, is_reporter, abuse_report=None):
+        """
+        Whether or not the decision can be appealed.
+        """
+        now = datetime.now()
+        base_criteria = (
+            self.date
+            and self.date >= now - timedelta(days=APPEAL_EXPIRATION_DAYS)
+            # Can never appeal an original decision that has been appealed and
+            # for which we already have a new decision. In some cases the
+            # appealed decision (new decision id) can be appealed by the author
+            # though (see below).
+            and not self.appealed_decision_already_made()
+        )
+        user_criteria = (
+            # Reporters can appeal decisions if they have a report and that
+            # report has no appeals yet (the decision itself might already be
+            # appealed - it can have multiple reporters). Note that we're only
+            # attaching the abuse report to the original job, not the appeal,
+            # by design. Reporters can never appeal an appeal job.
+            (
+                is_reporter
+                and abuse_report
+                and hasattr(self, 'cinder_job')
```

Comment:
yes, and I realized from thinking about this while away (bad PTO'ing!) that it wasn't correct anyway - an appeal on a proactive decision is still from a proactive review but it will have a cinder job.

Reply:
https://github.com/mozilla/addons-server/pull/22052/commits/2de825f67faa2fc273f3b8273b58a93246e578eb